keanu-project/src/main/java/io/improbable/keanu/vertices/bool/BoolVertex.java

```diff
@@ -91,6 +93,24 @@ public boolean getValue(int... index) {
         return getValue().getValue(index);
     }
 
+    public BooleanTensor sampleScalarValuesAsTensor(int[] shape, KeanuRandom random) {
+        if (!TensorShape.isScalar(this.getShape())) {
```

Comment:
if (!this.getValue().isScalar())

Reply:
Can't `getValue` return `null` in some circumstances?

Reply:
Hmm good question. Also, it has side effects as it might call `lazyEval()`.

So your way is better. I wonder if we should wrap this as `this.isScalar()` so that it becomes the obvious choice.